test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusSerializerTests.cs

```diff
@@ -383,5 +384,290 @@ public void HistogramNaN()
                     + "$").Replace('\'', '"'),
                 Encoding.UTF8.GetString(buffer, 0, cursor));
         }
+
+        [Fact]
+        public void NullValueAttributeIsDropped()
+        {
+            var buffer = new byte[85000];
+            var metrics = new List<Metric>();
+
+            using var meter = new Meter(Utils.GetCurrentMethodName());
+            using var provider = Sdk.CreateMeterProviderBuilder()
+                .AddMeter(meter.Name)
+                .AddInMemoryExporter(metrics)
+                .Build();
+
+            var tags = new KeyValuePair<string, object>[]
+            {
+                new KeyValuePair<string, object>("key", null),
+            };
+
+            var counter = meter.CreateCounter<double>("test_counter");
+            counter.Add(10, tags);
+
+            provider.ForceFlush();
+
+            var cursor = PrometheusSerializer.WriteMetric(buffer, 0, metrics[0]);
+            Assert.Matches(
+                ("^"
+                    + "# TYPE test_counter counter\n"
+                    + "test_counter 10 \\d+\n"
```

Comment:
Attribute is dropped. Is that what we want?

Reply:
I feel either empty string or dropped would be fine.
Anyways we'll have to stick with the spec once it becomes Stable (currently the Prometheus workgroup is still hammering out the rules).

Reply:
I stuck with dropping it for now. Though I also changed it to drop empty string valued attributes as well since I suspect it makes sense to handle both of these scenarios the same.